packages/netlify-cms-backend-gitlab/src/implementation.js

```diff
@@ -174,4 +181,61 @@ export default class GitLab {
       cursor: newCursor,
     }));
   }
+
+  // Editorial Workflow
+
+  unpublishedEntries() {
+    return this.api.listUnpublishedBranches().then(branches => {
+      const sem = semaphore(MAX_CONCURRENT_DOWNLOADS);
+      const promises = [];
+      branches.forEach(branch => {
+        promises.push(
+          new Promise(resolve => {
+            // Strip the `CMS_BRANCH_PREFIX` from the `branch.name` to get the `slug`.
+            const slug = branch.name.split(CMS_BRANCH_PREFIX).pop();
+            return sem.take(() =>
+              this.unpublishedEntry(null, slug)
```

Comment:
I assumed that `collection` is not used in `unpublishedEntry`, but we probably should infer it from the `branch`.